core/src/main/java/org/jdbi/v3/core/config/ConfigRegistry.java

```diff
@@ -81,12 +81,14 @@ public <C extends JdbiConfig<C>> C get(Class<C> configClass) {
     }
 
     private Function<ConfigRegistry, JdbiConfig<?>> configFactory(Class<? extends JdbiConfig<?>> configClass) {
-        return configFactories.computeIfAbsent(configClass, klass ->
-                registry -> {
-                    var config = JdbiClassUtils.findConstructorAndCreateInstance(klass, JDBI_CONFIG_TYPES, registry);
-                    config.setRegistry(registry);
-                    return config;
-                });
+        return configFactories.computeIfAbsent(configClass, klass -> {
+            var handleHolder = JdbiClassUtils.findConstructor(klass, JDBI_CONFIG_TYPES);
+            return registry -> {
+                var config = (JdbiConfig<?>) handleHolder.invoke(handle -> handle.invokeExact(registry));
```

Comment:
I tried that, but it requires passing an explicit return type as a method parameter to do a `methodHandle.asType(methodHandle.type().changeReturnType(<that_type>));` so that the lambda would be `handle -> (<that_type> handle.invokeExact(...)`. I am not sure that dragging around an additional parameter at all call sites, which is almost always the same as the type (except when it is not, namely here) makes the code more readable.

It is just `(JdbiConfig<?>) invoke()` because the return type of the c'tor is cast to be Object so that the invokeExact with Object return type works (so there is no extra cast). At least that is what I think, I am still pretty shaky with the method handles.